server/questions.js

```diff
@@ -18,149 +18,174 @@ You should have received a copy of the GNU General Public License
 
-var questions = db.collection('questions');
-exports.questions = questions;
-
-exports.QUESTION_REGULAR    = 0;
-exports.QUESTION_MULTCHOICE = 1;
-
-var nextid;
+/*
+ * Insert a new regular question into the database.
+ * The question object passed to the function should have
+ * the text, topic, type, answer, points and hint set.
+ */
+exports.addQuestionByTypeWithRedirection = function(qType, question, callback) {
+	switch(qType){
+		case common.questionTypes.REGULAR:
+			addRegularQuestion(question, callback);
+			break;
+		case common.questionTypes.MULTIPLECHOICE:
+			addMultipleChoiceQuestion(question, callback);
+			break;
+		default:
+			callback('failure', null);
+	}
+}
 
-/* Initialize the next question ID. */
-exports.questionInit = function(callback) {
-    questions.count(function(err, num) {
-        if (err) {
-            logger.error(err);
-            process.exit(1);
-        }
+exports.addRegularQuestion = function(question, callback) {	addRegularQuestion(question, callback); }
+exports.addMultipleChoiceQuestion = function(question, callback) {	addMultipleChoiceQuestion(question, callback); }
+
+var addRegularQuestion = function(question, callback) {
+	var currentDate = new Date().toString();
+	var questionToAdd = {};
+	questionToAdd.topic = question.topic;
+	questionToAdd.title = question.title;
+	questionToAdd.text = question.text;
+	questionToAdd.answer = question.answer;
+	questionToAdd.hint = question.hint;
+	questionToAdd.points = question.points;
+	questionToAdd.type = common.questionTypes.REGULAR;
+	questionToAdd.attempted = [];
+	questionToAdd.answered = [];
+	questionToAdd.attempts = [];
+	questionToAdd.ctime = currentDate;
+	questionToAdd.mtime = currentDate;
+	db.addRegularQuestion(questionToAdd, function(err, res) {
+		callback(err, res);
+	});
+}
 
-        if (num == 0) {
-            nextid = 0;
-            callback(nextid);
-        } else {
-            /* get the maximum question id in the database */
-            questions.find().sort({id: -1}).limit(1).toArray(function(err, docs) {
-                if (err) {
-                    logger.error(err);
-                    process.exit(1);
-                }
-                nextid = docs[0].id + 1;
-                callback(nextid);
-            });
-        }
-    });
+var addMultipleChoiceQuestion = function(question, callback) {
+	var currentDate = new Date().toString();
+	var questionToAdd = {};
+	questionToAdd.topic = question.topic;
+	questionToAdd.title = question.title;
+	questionToAdd.text = question.text;
+	questionToAdd.answer = question.answer;
+	questionToAdd.hint = question.hint;
+	questionToAdd.points = question.points;
+	questionToAdd.type = common.questionTypes.MULTIPLECHOICE;
+	questionToAdd.attempted = [];
+	questionToAdd.answered = [];
+	questionToAdd.attempts = [];
+	questionToAdd.ctime = currentDate;
+	questionToAdd.mtime = currentDate;
+	db.addMultipleChoiceQuestion(questionToAdd, function(err, res) {
+		callback(err, res);
+	});
```

Comment:
Can you add questionToAdd.answerOptions --> dictionary to store different answer options

Reply:
Will do